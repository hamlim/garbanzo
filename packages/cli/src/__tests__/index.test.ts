import { describe, expect, test } from "bun:test";
import { CLI, type DefaultOptions, type ParsedArgs } from "../index";

class TestLogger {
  logs: Array<any> = [];

  log(...messages: Array<any>) {
    this.logs.push(messages);
  }
}

describe("CLI", () => {
  test("should parse args", async () => {
    let testLogger = new TestLogger();

    let cli = new CLI(
      { _: ["help"] } as ParsedArgs,
      {
        logger: testLogger,
      } as DefaultOptions,
    );

    await cli.run();

    let logs = testLogger.logs.join("\n");

    expect(logs).toMatchSnapshot();
  });
});
