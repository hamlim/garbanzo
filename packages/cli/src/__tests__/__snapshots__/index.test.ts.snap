// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI should parse args 1`] = `
"@garbanzo/cli:

Usage:
  garbanzo init  - Initialize a new Garbanzo project
  garbanzo dev   - Start the development server
  garbanzo build - Build the project
  garbanzo help  - Show this help message
"
`;
